cw-orch-daemon/src/senders/sender_trait.rs

```diff
@@ -0,0 +1,52 @@
+use cosmrs::{tx::Msg, AccountId, Any};
+use cosmwasm_std::Addr;
+use cw_orch_core::environment::ChainInfoOwned;
+use tonic::transport::Channel;
+
+use crate::{CosmTxResponse, DaemonError};
+
+pub trait SenderTrait: Clone {
+    type Error: Into<DaemonError> + std::error::Error + std::fmt::Debug + Send + Sync + 'static;
+    type SenderOptions: Default + Clone;
+
+    // TODO: do we want to enforce sync on this function ?
+    fn address(&self) -> Result<Addr, Self::Error>;
```

Comment:
Is there a case where we would prefer async? Multisigner?

Reply:
I don't think so for now. Let's think about this later when we need it ? Removing the todo